### PR TITLE
enhancement(config): Add acknowledgement config to globals

### LINKED ISSUE #10356: A global setting to enable/disable end-to-end acknowledgements
Instead of opting into end-to-end acknowledgements within each individual source, it would be nice to opt-in globally for organizations that broadly want durability. For example:

```toml
# vector.toml
acknowledgements = true
```

### PR DESCRIPTION
**Pull request by bruceg**
Closes #10356

I'd like to add tests akin to the `generate_config` test, but am having difficulty coming up with something simple that makes sure this pattern is followed.

### CODE REVIEW

File: website/cue/reference/configuration.cue [hunk 1]
```diff
@@ -7,6 +7,21 @@ configuration: {
 
 configuration: {
 	configuration: {
+		_acknowledgements: {
```
Comment (binarylogic) [inline website/cue/reference/configuration.cue:10]: Why the `_` prefix?

**Comment by binarylogic** [comment]
I wish we could test this. Almost, without fail, we have problems with global defaults like this. See #9115.

**Review by fuchsnj — APPROVED** [review]

**Comment by bruceg** [comment]
@binarylogic @jszwedko Where are we at with this? Do we want this global setting to allow users to enable the feature across the board, or should this be dropped?

**Comment by jszwedko** [comment]
I'm personally OK with punting on this in-lieu of the other e2e work which I think is more blocking. If we do add a global option, we should ensure that it can be disabled for certain sources, as you suggested. This sort of thing is regrettably hard to test 😞

**Comment by bruceg** [comment]
The changes in 0e973e5 now allow for individual sources to override the global setting.

File: website/cue/reference/configuration.cue [hunk 2]
```diff
@@ -7,6 +7,21 @@ configuration: {
...
 configuration: {
 	configuration: {
+		acknowledgements: {
+			common:      true
+			description: "Controls how acknowledgements are handled by all sources. These settings may be overridden in individual sources."
+			required:    false
+			type: object: options: {
+				enabled: {
+					common:      true
+					description: "Controls if sources will wait for destination sinks to deliver the events before acknowledging receipt. If set to `true`, all capable sources will have acknowledgements enabled."
```
Comment (jszwedko) [inline website/cue/reference/configuration.cue:17]:
```suggestion
					description: "Controls if sources will wait for destination sinks to deliver the events, or persist them to a disk buffer, before acknowledging receipt. If set to `true`, all capable sources will have acknowledgements enabled."
```

**Review by jszwedko — APPROVED** [review]
Aha, yeah, I see that now. I do, again, wish we could test these sorts of global + local configuration, but I'm ok with it going in as-is given it keeps with precedent.

File: website/cue/reference/configuration.cue (hunk 2 above)
Comment (jszwedko) [reply website/cue/reference/configuration.cue:17]: Might also be worth linking to the acknowledgement docs here once they exist too.

Comment (bruceg) [reply website/cue/reference/configuration.cue:17]: Agreed.

**Review by tobz — APPROVED** [review]
This looks good to me, save for the outstanding duplicate import that should be fixed.

File: lib/vector-core/src/config/global_options.rs [hunk 3]
```diff
@@ -4,7 +4,8 @@ use serde::{Deserialize, Serialize};
 use shared::TimeZone;
 use snafu::{ResultExt, Snafu};
 
-use crate::config::{proxy::ProxyConfig, LogSchema};
+use super::{proxy::ProxyConfig, AcknowledgementsConfig, LogSchema};
+use crate::config::{proxy::ProxyConfig, serde::bool_or_struct, LogSchema};
```
[the remark this answers is not in the document]
Comment (bruceg) [reply lib/vector-core/src/config/global_options.rs]: Done in 8070c89

**Review by tobz — APPROVED** [review]

**Comment by bruceg** [comment]
This PR seems to have changed the order in which config fields are serialized, as evidenced in 9efe629. The comments in `src/config/builder.rs` seem to imply this may be a breaking change. Is this a concern?

**Comment by bruceg** [comment]
Never mind, it must have been an unintended change.

**Pull request merged**